src/matter/cluster/OnOffCluster.ts

```diff
@@ -4,18 +4,109 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { BooleanT } from "../../codec/TlvObjectCodec";
-import { Attribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { BooleanT, Field, ObjectT, StringT, Template, UnsignedIntT, UnsignedLongT } from "../../codec/TlvObjectCodec";
+import { Attribute, OptionalAttribute, OptionalWritableAttribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { TlvType } from "../../codec/TlvCodec";
 
+export const enum OnOffStartUpOnOff {
+    Off = 0,
+    On = 1,
+    Toggle = 2,
+}
+const OnOffStartUpOnOffT = { tlvType: TlvType.UnsignedInt } as Template<OnOffStartUpOnOff>;
+
+export const enum OnOffEffectIdentifier {
+    DelayedAllOff = 0,
+    DyingLight = 1,
+}
+const OnOffEffectIdentifierT = { tlvType: TlvType.UnsignedInt } as Template<OnOffEffectIdentifier>;
+
+export const enum OnOffDelayedAllOffEffectVariant {
+    FadeToOffIn_0p8Seconds = 0,
+    NoFade = 1,
+    "50PercentDimDownIn_0p8SecondsThenFadeToOffIn_12Seconds" = 2,
+}
+const OnOffDelayedAllOffEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDelayedAllOffEffectVariant>;
+
+export const enum OnOffDyingLightEffectVariant {
+    "20PercentDimUpIn_0p5SecondsThenFadeToOffIn_1Second" = 0,
+}
+const OnOffDyingLightEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDyingLightEffectVariant>;
+
+/*
+  <bitmap name="OnOffControl" type="BITMAP8">
+    <cluster code="0x0006"/>
+    <field name="AcceptOnlyWhenOn" mask="0x01"/>
+  </bitmap>
+*/
+
+const OffWithEffectRequestT = ObjectT({
+    effectId: Field(0, OnOffEffectIdentifierT),
+    effectVariant: Field(1, OnOffDelayedAllOffEffectVariantT), /* OR OnOffDyingLightEffectVariantT depending on first field */
+});
+
+const OnWithTimedOffRequestT = ObjectT({
+    onOffControl: Field(0, UnsignedIntT), /* TODO: Type is a Bit map! */
+    onTime: Field(1, UnsignedIntT), /* min:0, max: 254 */
+    offWaitTime: Field(2, UnsignedIntT), /* min:0, max: 254 */
+});
+
+/**
+ * Attributes and commands for switching devices between 'On' and 'Off' states.
+ */
 export const OnOffCluster = Cluster(
     0x06,
     "On/Off",
     {
-        on: Attribute(0, BooleanT),
+        onOff: Attribute(0, BooleanT, false), /* reportable: true */
+        globalSceneControl: OptionalAttribute(0x4000, BooleanT, true),
+        onTime: OptionalWritableAttribute(0x4001, UnsignedIntT, 0),
+        offWaitTime: OptionalWritableAttribute(0x4002, UnsignedIntT, 0),
+        startUpOnOff: OptionalWritableAttribute(0x4003, OnOffStartUpOnOffT), /* nullable: true, writeAcl: manage */
```

Comment:
We also need a "nullable" because else noone could set the value to "null" when using the Cluster and null is a valid value in this case: 

From specs:
> If the value is null, the OnOff attribute is set to its previous value.

Reply:
Ok, will tell when I stunmble over it in a more relevant usecase :-)